### PR TITLE
feat: add payment or receipt submission feature

### PR DESCRIPTION
**Pull request by Srish-ty**
- Adding bank transaction details
- Adding UPI QR code for payment
- A submission portal for receipt or screenshot after payment as proof of payment
- function to upload file to cloudinary
- find doc using userID and update with the proofURL

### CODE REVIEW

**Comment by Srish-ty** [comment]
@jayeshxn , @Shurtu-gal please merge

File: src/components/payment/Final.jsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+function Final() {
+	const { userInfo } = useContext(AuthContext);
+	var currentUser = userInfo[0];
+
+	const [file, setFile] = useState(null);
+
+	const uploadProof = async () => {
+		const proofURL = await uploadReceipt(file);
+		await updateDocumentByUid(currentUser.uid, proofURL);
+		await updateDocumentByUid(currentUser.uid, proofURL);
```
Comment (Shurtu-gal) [inline src/components/payment/Final.jsx:18]: Do we need this 2 times?

File: src/components/payment/Final.jsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+import { AuthContext } from '../../context/AuthContext';
+import { PaymentCont, SubmissionForm, UPI } from './Payment';
+import { Heading, Paragraph } from '../shared';
+import { Contactorg, Emaildata } from '../../data/paymentData';
+import { uploadReceipt } from './uploadReceipt';
+import { updateDocumentByUid } from '../../firebase/uploadProof';
+
+function Final() {
+	const { userInfo } = useContext(AuthContext);
+	var currentUser = userInfo[0];
```
Comment (Shurtu-gal) [inline src/components/payment/Final.jsx:11]: nit: can be const as is not being changed I believe.

Furthermore please show some error message using toast and redirect if user is not logged in.

File: src/components/payment/uploadReceipt.js
Comment (Shurtu-gal) [inline src/components/payment/uploadReceipt.js:1]: If I am not wrong there was already an uploadToCloudinary function. Is this any different.

**Review by Shurtu-gal — CHANGES_REQUESTED** [review]

File: src/components/payment/uploadReceipt.js
Comment (Srish-ty) [reply src/components/payment/uploadReceipt.js:1]: Yeah. Have two different presets, one for profile-images and another for proof of payments.

Comment (Shurtu-gal) [reply src/components/payment/uploadReceipt.js:1]: Okay

File: src/components/payment/Final.jsx (hunk 1 above)
Comment (Srish-ty) [reply src/components/payment/Final.jsx]: removed . https://github.com/dscnitrourkela/project-udon/pull/33/commits/82dcf6ed3539e304f203451f64e4ef473a50a7cd

**Comment by Srish-ty** [comment]
@Shurtu-gal , please review

**Comment by Srish-ty** [comment]
@Shurtu-gal , please review

**Review by Shurtu-gal — APPROVED** [review]
Great work @Srish-ty !! 🚀

**Pull request merged**